distributed/worker.py

```diff
@@ -298,7 +298,11 @@ def _close(self, report=True, timeout=10):
             v.stop()
         self.rpc.close()
         self.status = 'closed'
-        self.stop()
+        with ignoring(gen.TimeoutError):
+            yield gen.with_timeout(timedelta(seconds=1),
+                    [close(stream) for stream in self._listen_streams])
+        for stream in self._listen_streams:
+            stream.close()
```

Comment:
The problem persists.  It also persists if you remove the calls to `close(stream)` entirely and just call `stream.close()`.